tests/projects/test_project.py

```diff
@@ -398,7 +398,7 @@ def test_load_project_and_sync_functions(
     assert len(project.spec._function_objects) == expected_num_of_funcs
 
     if sync:
-        function_names = project.get_function_names()
+        function_names = project.spec._function_definitions.keys()
```

Comment:
looking at the `get_function_names` is seems to put the `function` in the list of the names instead of the key when the `function` is of type string, I am not sure when this happens but note that it can have different output

Reply:
True, it is left for external use (not used internally anymore - we should be using the keys here and not the names)